pkg/controllers/allocation/binpacking/packer.go

```diff
@@ -63,6 +81,14 @@ type Packing struct {
 // It follows the First Fit Decreasing bin packing technique, reference-
 // https://en.wikipedia.org/wiki/Bin_packing_problem#First_Fit_Decreasing_(FFD)
 func (p *packer) Pack(ctx context.Context, schedule *scheduling.Schedule, instances []cloudprovider.InstanceType) []*Packing {
+	startTime := time.Now()
+	packings := p.pack(ctx, schedule, instances)
+	packTimeHistogram.Observe(time.Since(startTime).Seconds())
+
+	return packings
+}
+
+func (p *packer) pack(ctx context.Context, schedule *scheduling.Schedule, instances []cloudprovider.InstanceType) []*Packing {
```

Comment:
To remove the need for a private function that just calls the public, you could place this at the top of original and still make minimal code additions to the core packing logic.  

Pack:
```
startTime := time.Now() 
defer packTimeHistogram.Observe(time.Since(startTime).Seconds())
```

The `defer` is advantageous on the Observe call anyways since you'll get observed durations even if a panic or some runtime errors wreak havoc.

Reply:
> To remove the need for a private function that just calls the public [...]

It's the other way around: the public method calls the private method.  The wrapper allows separating the metric logic from the binpacking logic.  Is there an advantage to breaking the separation of concerns?

Reply:
yeah, sorry mistyped: Public calling Private. 

Since the metric logic is tiny (two lines), seems a bit heavy to do the function proxying. The metric logic can be placed at the top with the defer to keep it together to separate it a bit for the reader.  The reason to keep the private without metrics is if we have a use-case (maybe in testing) to not emit metrics and use just the `pack` func. But that doesn't seem like a great thing to do anyways. I would guess it would be configurable elsewhere to make the Observe func a noop, if we don't want metrics.